src/collectors/nvidia_gpu/nvidia_gpu.py

```diff
@@ -0,0 +1,69 @@
+# coding=utf-8
+
+"""
+The NvidiaGPUCollector collects GPU utilization metric using nvidia-smi.
+
+#### Dependencies
+
+ * /usr/bin/nvidia-smi
+
+"""
+
+from itertools import izip
+
+import diamond.collector
+
+
+class NvidiaGPUCollector(diamond.collector.ProcessCollector):
+
+    def get_default_config_help(self):
+        config_help = super(NvidiaGPUCollector, self).get_default_config_help()
+        config_help.update({
+            'bin': 'The path to the ping binary'
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+        config = super(NvidiaGPUCollector, self).get_default_config()
+        config.update({
+            'path': 'nvidia',
+            'bin': '/usr/bin/nvidia-smi',
+            'stats': [
+                'index',
+                'memory.total',
+                'memory.used',
+                'memory.free',
+                'utilization.gpu',
+                'utilization.memory',
+                'temperature.gpu'
+            ]
+        })
+        return config
+
+    def collect(self):
+        """
+        Collector GPU stats
+
+        """
+
+        stats_config = self.config['stats']
+        raw_output = self.run_command([
+            '--query-gpu={query_gpu}'.format(query_gpu=','.join(stats_config)),
+            '--format=csv,nounits,noheader'
+        ])
+        if raw_output is None:
+            return
+        results = raw_output[0].strip().split("\n")
+        for result in results:
+            stats = result.strip().split(',')
+            assert len(stats) == len(stats_config)
+            index = stats[0]
+            for stat_name, metric in izip(stats_config[1:], stats[1:]):
+                metric_name = 'gpu_{index}.{stat_name}'.format(
```

Comment:
Should the metric name be prefixed with gpu_ ? Is this a common naming convention or can there be other types of objects?